src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -271,26 +280,119 @@ private BoundInterpolatedString BindUnconvertedInterpolatedStringToHandlerType(B
+            {
+                // MakeConstructorInvocation can call CoerceArguments on the builder if overload resolution succeeded ignoring accessibility, which
+                // could still end up not succeeding, and that would end up changing the arguments. So we want to clear and repopulate.
+                argumentsBuilder.Clear();
+
+                // Try again with an out parameter. Note that we intentionally use `diagnostics` for resolving System.Boolean, because we want to
+                // track that we're using the type no matter what.
+                var boolType = GetSpecialType(SpecialType.System_Boolean, diagnostics, unconvertedInterpolatedString.Syntax);
+                var trailingConstructorValidityPlaceholder = new BoundInterpolatedStringArgumentPlaceholder(unconvertedInterpolatedString.Syntax, BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter, boolType);
+                var outConstructorAdditionalArguments = additionalConstructorArguments.NullToEmpty().Add(trailingConstructorValidityPlaceholder);
+                populateArguments(unconvertedInterpolatedString, outConstructorAdditionalArguments, baseStringLength, numFormatHoles, intType, argumentsBuilder);
+
+                refKindsBuilder.Add(RefKind.Out);
+
+                var outConstructorDiagnostics = BindingDiagnosticBag.GetInstance(template: diagnostics);
+                var outConstructorCall = MakeConstructorInvocation(interpolatedStringHandlerType, argumentsBuilder, refKindsBuilder, unconvertedInterpolatedString.Syntax, outConstructorDiagnostics);
+
+                if (outConstructorCall is BoundObjectCreationExpression { ResultKind: LookupResultKind.Viable })
+                {
+                    // We successfully bound the out version, so set all the final data based on that binding
+                    constructorCall = outConstructorCall;
+                    diagnostics.AddRangeAndFree(outConstructorDiagnostics);
+                    nonOutConstructorDiagnostics.Free();
+                    additionalConstructorArguments = outConstructorAdditionalArguments;
+                }
+                else
+                {
+                    // We'll attempt to figure out which failure was "best" by looking to see if one failed to bind because it couldn't find
+                    // a constructor with the correct number of arguments. We presume that, if one failed for this reason and the other failed
+                    // for a different reason, that different reason is the one the user will want to know about. If both or neither failed
+                    // because of this error, we'll report everything.
+
+                    var nonOutConstructorHasArityError = nonOutConstructorDiagnostics.DiagnosticBag?.AsEnumerableWithoutResolution().Any(d => (ErrorCode)d.Code == ErrorCode.ERR_BadCtorArgCount) ?? false;
```

Comment:
>var nonOutConstructorHasArityError = nonOutConstructorDiagnostics.DiagnosticBag?.AsEnumerableWithoutResolution().Any(d => (ErrorCode)d.Code == ErrorCode.ERR_BadCtorArgCount) ?? false; [](http://example.com/codeflow?start=20&length=183)

I think I agree with Chris, this feels fragile. We should probably do the analysis based on the failure kind for candidates instead, and use the same logic to deside which ones are the "best" among "bad" as we use to decide what errors to report when overload resolution failed. It probably should be logically similar to performing overload resolution on both sets combined at the same time only some candidates allow omitted out argument.